### PR TITLE
Delete pods to restart KMP

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
Changing Deployment replicas to restart KMP does not work well with CNAO
tier1 tests since CNAO will Reconcile the Deployment after changing the
replicas so it's not always working. This change replace the restart
mechanism with KMP pods deletiong and deployment condition check.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/tests.go [hunk 1]
```diff
@@ -315,6 +306,40 @@ func changeReplicas(managerName string, numOfReplica int32) error {
...
+				if condition.Type == v1.DeploymentAvailable {
+					return condition.Status == conditionStatus
+				}
+			}
+
+			return true
+		}, 2*time.Minute, time.Second/10).Should(BeTrue(), "Failed waiting readiness at for deployment:\n%v", string(deploymentName))
+	}
+
+	deploymentConditionAvailability(corev1.ConditionFalse, 2*time.Minute, time.Second/10)
```
Comment (RamLavi) [inline tests/tests.go:338]: What if the deployment is ready before getting to this line?

**Review by RamLavi — COMMENTED** [review]
I'm concerned that now that KMP pods are ready faster, that polling the deployment for readiness will result with flakiness...

File: tests/tests.go (hunk 1 above)
Comment (qinqon) [reply tests/tests.go:338]: restarting pods takes enouhg time to reach this, or this is what I have see on multiple runs

I can go back to check replicas but we can face the same problem and enmascarade a false Readiness.

Comment (RamLavi) [reply tests/tests.go:338]: How do you feel about waiting for the "old" pods to be cleared by specific-name, before changing the deployment replicas back to 1?

Comment (qinqon) [reply tests/tests.go:338]: Ahh that's a good one :-) will do thanks!

File: tests/tests.go [hunk 2]
```diff
@@ -315,6 +306,40 @@ func changeReplicas(managerName string, numOfReplica int32) error {
 	return nil
 }
 
+func deletePodsFromDeployment(deploymentName string) error {
```
Comment (RamLavi) [inline tests/tests.go:309]: maybe `restartPodsFromDeployment` is more exact here?

File: tests/tests.go (hunk 1 above)
Comment (qinqon) [reply tests/tests.go]: Done

File: tests/tests.go (hunk 2 above)
Comment (qinqon) [reply tests/tests.go]: Done

**Review by RamLavi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**